### DISCUSSION TITLE
Getting Image pose

### DISCUSSION THREAD

**Discussion by Inshu32** [stella-cv/stella_vslam#618] [Q&A] [answered]
Hi all,

I have a probem in hand in which I am running Stellla- V slam on a 360 degree video. It is working wonderfully and I am getting the tracked path correctly. Now I want to extract camera/ image pose at some particular timestamps. I have already extracted the timestaps which are of my interest, now how to get the image poses at these timestamps is the question. Any leads would be helpful. Thanks in Advance.

**Comment by Inshu32** [comment]
Does the following line of code works for the Stella VSLAM latest version: 

auto map_publisher_ = SLAM.get_map_publisher();
auto camera_pose = map_publisher_->get_current_cam_pose().inverse().eval(); 

I can get pose from this command ?

**Reply by Inshu32** [reply]
Cannot find the Run_vslam.cc file where I can add the above 2 variables.

**Reply by Inshu32** [reply]
@Skwangles

**Reply by Skwangles** [reply]
Run_***_slam is here:
https://github.com/stella-cv/stella_vslam_examples

Example docs here:
https://stella-cv.readthedocs.io/en/latest/example.html

Map publisher header file: https://github.com/stella-cv/stella_vslam/blob/main/src%2Fstella_vslam%2Fpublish%2Fmap_publisher.h

**Reply by Inshu32** [reply]
@Skwangles Thanks a lot. One more question, What does map.msg file contain.. unable to find this information in documentation. Can I get Image pose directly from this database file ?

**Reply by Skwangles** [reply, marked as answer]
Here's the format: https://msgpack.org/index.html

I'd recommend switching to using 'sqlite3' (replace 'msgpack' in the config yaml file with 'sqlite3') if you wanted to do that. It has the pose of each keyframe in a 4x4 pose matrix.

**Reply by Inshu32** [reply]
--map-db-out map_file_name.msg option. what is the extension of sqlite3 ? like msg file is saved with .msg extension.

**Reply by Skwangles** [reply]
A quick google is always helpful for these types of questions
See https://en.wikipedia.org/wiki/SQLite#:~:text=.sqlite%2C%20.sqlite3%2C%20.db%2C%20.db3%2C%20.s3db%2C%20.sl3
